src/miner.h

```diff
@@ -89,7 +98,7 @@ CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddre
 
 #ifdef ENABLE_MINING
 /** Get -mineraddress */
-void GetMinerAddress(MinerAddress &minerAddress);
+void GetMinerAddress(std::optional<MinerAddress> &minerAddress);
```

Comment:
This is now a pretty weird API; I think that we should at very least document what it means to pass std::nullopt here, or preferably change the API to take no arguments and return `std::optional<MinerAddress>` instead.

Reply:
The API here is an artifact of how the C++ signals being used by the Bitcoin code work. Each registered listener to a signal is passed the given arguments in turn, and in this case each has a chance at modifying them.

Reply:
Hmm, looking at https://www.boost.org/doc/libs/1_78_0/doc/html/signals2/tutorial.html#id-1.3.37.4.5.3 it appears that it _is_ possible to change the signals to directly return values, by implementing a "combiner" (the default combiner just returns the last signal's return value, which is clearly wrong). This is doable but a more invasive change, so I've opened #5631 for it.

Reply:
Ok, I wasn't sure how this signature interacted with the async system; the current is fine.